### ISSUE TITLE
Allow obtaining original string value from ConvertFrom-Json returned DateTime

### ISSUE THREAD

**Issue by dancojocaru2000** [PowerShell/PowerShell#16976]
### Summary of the new feature / enhancement

As a user I want to be able to access the original string value of a converted DateTime so that it can be used in cases where an equivalent ISO 8601 representation would be considered a different value.

I am using an API which returns ISO 8601 strings as IDs. However, PowerShell automatically converts those IDs to DateTime, and using something like `Get-Date -Format o $date` returns a different string that results in a different, invalid ID.

### Proposed technical implementation details (optional)

There are two ways I can see this implemented:

1. A `-NoDateConversion` flag to `Convert-FromJson` and cmdlets that use it such as `Invoke-RestMethod`
2. The returned DateTime could have an `OriginalString` member, so that the original string can be accessed like `$date.OriginalString`
3. The returned DateTime, when passed to `Get-Date -Format o $date`, would output the original string instead of a ISO 8601 equivalent reinterpretation.

**Comment by dwtaber** [comment]
Maybe this concept should be applied to other types the cmdlet converts from strings as well, rather than making `DateTime` a special case.

**Comment by iSazonov** [comment]
Please repro steps and input file example.

**Comment by dancojocaru2000** [comment]
> Maybe this concept should be applied to other types the cmdlet converts from strings as well, rather than making `DateTime` a special case.

Agreed, DateTime was the first to get me in trouble so that's why I singled it out.

**Comment by dancojocaru2000** [comment]
> Please repro steps and input file example.

```pwsh
$json = "[{`"id`": `"2022-03-09T09:11:04.254630+00:00`", `"otherData`": true}]"
$converted = ConvertFrom-Json $json
$converted[0].id | Get-Date -Format o
```

As can be seen, the input string is `2022-03-09T09:11:04.254630+00:00`. The output of the last command on my machine is `2022-03-09T11:11:04.2546300+02:00`. Since the two strings aren't identical, using the 2nd one as the ID results in an error.

I want to highlight that I don't want to obtain a `+00:00` ISO 8601 string! It just so happens that this time the input is `+00:00`. If the input would be `2022-03-09T11:11:04.2546300+02:00` and the output would be `2022-03-09T09:11:04.254630+00:00`, it would still be an error.

**Comment by iSazonov** [comment]
I see ConvertFrom-Json converts the string to _local_ datetime. It is how underlying NewtonSoft.NET works. So we can not change this. You have to do some manual steps to convert local datetime to string you need.

**Comment by dancojocaru2000** [comment]
Is it not possible to somehow disable the automatic conversions and keep all strings as strings? Because this seems as a flaw to me (that should be perhaps fixed in NewtonSoft.NET), destructively converting data without an option to get the original back.

**Comment by iSazonov** [comment]
> Edit: I found this Stack Overflow question: [Json.NET Disable the deserialization on DateTime](https://stackoverflow.com/questions/11856694/json-net-disable-the-deserialization-on-datetime). Would it not possible to pass such options?

No, it will be huge breaking change.

Is your source string always in UTC? If yes, you could use an workaround otherwise it is a dead end.

**Comment by dancojocaru2000** [comment]
Why would it be a breaking change?

And no, the source string is not always following a certainl format.

**Comment by iSazonov** [comment]
> Why would it be a breaking change?

The cmdlet will return different result.

> And no, the source string is not always following a certain format.

In the case you could pre-process the json so that "hide/mask" the datetime string.

**Comment by dancojocaru2000** [comment]
> > Why would it be a breaking change?
> 
> 
> 
> The cmdlet will return different result.

There could be a `-NoConversion` flag which would be a backwards compatible change.

**Comment by jborean93** [comment]
This is definitely possible to do without a breaking change. You can have an opt in parameter like `-DateTimeFormat None|DateTime|DateTimeOffset` that controls the [DateParseHandling](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_DateParseHandling.htm) property in Newtonsoft. PowerShell already https://github.com/PowerShell/PowerShell/blob/7dc4587014bfa22919c933607bf564f0ba53db2e/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs#L181-L187 passing in a settings class so it shouldn't be that hard to add. The only concern I would have is that it would tie `ConvertFrom-Json` more closely to `Newtonsoft` making it harder to potentially migrate to another JSON library in the future (if that's ever going to happen).

The current `DateTime` handling is a bit subpar because it always gives you a local kind `DateTime` object where you can loose information on the raw value like what TZ it was in. For example my local offset is +10:00 and this is what I see

```powershell
$json = '"2022-03-09T09:11:04.25463+05:00"'
$converted = ConvertFrom-Json $json

$converted

# Wednesday, 9 March 2022 2:11:04 pm

$converted.Kind
# Local
```

I have no way to know with this deserialized value that the raw string had a `+05:00` offset, I just know what the value is in UTC and local time.

**Comment by mklement0** [comment]
See also:

* Proposal: #13598

* Relate discussion: #13592

**Comment by kilasuit** [comment]
Reopening this for WG to review as I do think this is a bug as there shouldn't be a conversion unless the user requests one, which may deviate from current experience but would be reasonable to change going forward.

**Comment by JJatOracle** [comment]
It looks like, for 7.5.0 at least, this feature is implemented with the `-DateKind` flag. I still wildly disagree with the behavior of `Default` not matching `String`, but I think this issue can be closed for now.

**Comment by SteveL-MSFT** [comment]
As noted, this has been fixed in 7.5 with the `-DateKind` paramter
